specs/governance/governor.md

```diff
@@ -0,0 +1,702 @@
+This contract uses the `GovernanceToken` contract for voting power snapshots, and the `ProposalTypesConfigurator`
+for proposal types.
+
+## Constants
+
+| Constant          | Value                           | Description |
+| ----------------- | ------------------------------- | ----------- |
+| `BALLOT_TYPEHASH` |  `0x150214d74d59b7d1e90c73fc22ef3d991dd0a76b046543d4d80ab92d2a50328f` | The EIP-712 typehash for the ballot struct  |
+| `EXTENDED_BALLOT_TYPEHASH` |  `0x7f08F3095530B67CdF8466B7a923607944136Df0` | The EIP-712 typehash for the extended ballot struct |
+| `COUNTING_MODE` | `support=bravo&quorum=against,for,abstain&params=modules` | The configuration for supported values for `castVote` and how votes are counted |
+| `PERCENT_DIVISOR` |  `10000` | The maximum value of `quorum` and `approvalThreshold` for proposal types  |
+| `PROPOSAL_TYPES_CONFIGURATOR` |  `0x67ecA7B65Baf0342CE7fBf0AA15921524414C09f` | The address of the proposal types configurator contract  |
+| `VERSION` |  `2` | The version of the `Governor` contract  |
+| `TOKEN` |  `0x4200000000000000000000000000000000000042` | The address of the `GovernanceToken` contract  |
+
+## Interface
+
+### Core Functions
+
+#### `cancel`
+
+Cancels a proposal. A proposal MUST only be cancellable by the manager and ONLY while it's in an active state,
+i.e. not executed or cancelled already.
+
+```solidity
+function cancel(address[] memory _targets, uint256[] memory _values, bytes[] memory _calldatas, bytes32 _descriptionHash) external returns (uint256)
+```
+
+This function MUST emit the `ProposalCanceled` event and return the ID of the proposal that was cancelled.
+
+#### `cancelWithModule`
+
+Similar to `cancel`, but for proposals that were created with a module.
+
+```solidity
+function cancelWithModule(address _module, bytes memory _proposalData, bytes32 _descriptionHash) external returns (uint256)
+```
+
+This function MUST emit the `ProposalCanceled` event and return the ID of the proposal that was cancelled.
+
+#### `castVote`
+
+Cast a vote on a proposal.
+
+```solidity
+function castVote(uint256 _proposalId, uint8 _support) external returns (uint256)
+```
+
+This function MUST emit the `VoteCast` event and return the voting weight.
+
+#### `castVoteBySig`
+
+Cast a vote on a proposal using a signature.
+
+```solidity
+function castVoteBySig(uint256 _proposalId, uint8 _support, uint8 _v, bytes32 _r, bytes32 _s) external returns (uint256)
+```
+
+This function MUST emit the `VoteCast` event and return the voting weight.
+
+#### `castVoteWithReason`
+
+Cast a vote on a proposal with a reason.
+
+```solidity
+function castVoteWithReason(uint256 _proposalId, uint8 _support, string memory _reason) external returns (uint256)
+```
+
+This function MUST emit the `VoteCast` event and return the voting weight.
+
+#### `castVoteWithReasonAndParams`
+
+Cast a vote on a proposal with a reason and additional encoded parameters.
```

Comment:
In what case would the encoded parameters be used?

Reply:
specified in https://github.com/ethereum-optimism/specs/pull/315/commits/9a0fb5809772b4a6840fb165b1d4579658b6c7cb